### ISSUE TITLE
Exchange  State_start and State_end has a different result

### ISSUE THREAD

**Issue by githublth** [AIS-Bonn/TopiCo#15]
hi, when i have parameter like  as below:
![image](https://user-images.githubusercontent.com/25904480/129167954-97c7c9f0-9a7d-4f0e-b1c1-0ee50e9d206f.png)
the result is :
![image](https://user-images.githubusercontent.com/25904480/129168149-48677afd-adfd-4f71-a83a-dce3287129bb.png)
but  when i exchange the State_start and  Waypoints as this:
![image](https://user-images.githubusercontent.com/25904480/129168389-4cb8f955-e60f-4f17-8e8b-ae939bd9a623.png)
the result like this 
![image](https://user-images.githubusercontent.com/25904480/129168478-00733515-30ee-425b-80fc-9fa4551a92c0.png)

obviously, the second one costs more time, I hope it have a result as the first one.  is it normal?
Thanks!

**Comment by MariusBeul** [comment]
Yes, the behavior is expected. Simply exchanging the start and end state does not work. You also have to invert the velocity and acceleration. Imagine a trajectory starting in the origin and going to 1m. If you have a **positive** velocity, the trajectory is faster, since the target position is **positive**. But if you are starting at 1m and want to go to the origin, a **negative** initial velocity makes your trajectory faster. Thus, you have to invert the signs of the initial velocity, acceleration, etc. and also of the final velocity if there is one.
Thus, the mirrored form of `[0.0,0.1,0.0]` to `[1.0,0.0,0.0]` is `[1.0,-0.1,0.0]` to `[0.0,0.0,0.0]`
